### ISSUE TITLE
Better `JsonReaderException` error message

### ISSUE THREAD

**Issue by nilskp** [plokhotnyuk/jsoniter-scala#754]
Once in a while, when I F up, I get the familiar exception:
```
com.github.plokhotnyuk.jsoniter_scala.core.JsonReaderException: missing required field "id", offset: 0x00000a14, buf:
+----------+-------------------------------------------------+------------------+
|          |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f | 0123456789abcdef |
+----------+-------------------------------------------------+------------------+
```
But having a ton of case classes, I really don't know, quickly, which case class is failing. It would be immensely helpful if the error message could provide the name of the case class. I'm hoping this will be trivial to add.

**Comment by plokhotnyuk** [comment]
@nilskp I've tried to provide some solution but it is limited and does not solve the problem of pointing in general.

Are you getting an empty hex dump? If yes than it is a bug and should be fixed. Do you have possibility to print a whole message in case of parsing error?

**Comment by nilskp** [comment]
@plokhotnyuk No, I'm not getting an empty hex dump; just didn't want to include it here, but it is printed and is sometimes helpful, but only when I know where to look.

**Comment by plokhotnyuk** [comment]
@nilskp I've added an ability to customize size of the hex dump: https://github.com/plokhotnyuk/jsoniter-scala/commit/1376558d68e3e9253f5d25204169cac64cd0bdec

Will it be helpful for your case?

**Comment by nilskp** [comment]
@plokhotnyuk Possibly, although I cannot remember a case where that would have been the case. Still can't hurt.

**Comment by plokhotnyuk** [comment]
Hi, @winitzki ! Thanks for trying jsoniter-scala.

Both configuration options: for increasing dump size and for turning on stack traces are in [`ReaderConfig`](https://github.com/plokhotnyuk/jsoniter-scala/blob/master/jsoniter-scala-core/shared/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/core/ReaderConfig.scala).

You can pass it's preconfigured instanse as a parameter for each `read...` call.

**Comment by winitzki** [comment]
Thank you, will try!

**Comment by plokhotnyuk** [comment]
@winitzki to make stack traces more helpful you can add the following implicit val that will turn on printing of derived codecs in the definition scope:
```
implicit val printCodec: CodecMakerConfig.PrintCodec = new CodecMakerConfig.PrintCodec {}
```
